profile_bluesky/startup/instrument/devices/lakeshores.py

```diff
@@ -0,0 +1,284 @@
+        self.setpoint.subscribe(self.done.get)
+
+    @property
+    def settle_time(self):
+        return self._settle_time
+
+    @settle_time.setter
+    def settle_time(self,value):
+        if value < 0:
+            raise ValueError('Settle value needs to be >= 0.')
+        else:
+            self._settle_time = value
+
+    @property
+    def tolerance(self):
+        return self._tolerance
+
+    @tolerance.setter
+    def tolerance(self,value):
+        if value < 0:
+            raise ValueError('Tolerance needs to be >= 0.')
+        else:
+            self._tolerance = value
+            _ = self.done.get()
+
+    @property
+    def egu(self):
+        return self.units.get(as_string=True)
+
+    def stop(self,*,success=False):
+        if success is False:
+            self.setpoint.put(self._position)
+        super().stop(success=success)
+
+    def pause(self):
+        self.setpoint.put(self._position)
+
+    @done.sub_value
+    def _move_changed(self,**kwargs):
+        super()._move_changed(**kwargs)
+
+    def move(self,*args,**kwargs):
+        # TODO: This self.done.put(0) is for the cases where the end point is
+        #within self.tolerance. Is it needed? Or is there a better way to do this?
+        self.done.put(0)
+        return super().move(*args,**kwargs)
+
+class LS336_LoopRO(Device):
+    """
+    Additional controls for loop1 and loop2: heater and pid
+    """
+    readback = FormattedComponent(EpicsSignalRO, "{self.prefix}IN{self.loop_number}",
+                                  kind=Kind.hinted)
+    units = FormattedComponent(EpicsSignalWithRBV, "{self.prefix}IN{self.loop_number}:Units",
+                               kind=Kind.omitted)
+    loop_name = FormattedComponent(EpicsSignalRO,"{self.prefix}IN{self.loop_number}:Name_RBV",
+                                   kind=Kind.omitted)
+
+    def __init__(self, *args,loop_number=None,**kwargs):
+        self.loop_number = loop_number
+        super().__init__(*args,**kwargs)
+
+class LS336Device(Device):
+    """
+    support for Lakeshore 336 temperature controller
+    """
+    loop1 = FormattedComponent(LS336_LoopControl, "{self.prefix}", loop_number=1)
+    loop2 = FormattedComponent(LS336_LoopControl, "{self.prefix}", loop_number=2)
+    loop3 = FormattedComponent(LS336_LoopRO, "{self.prefix}", loop_number=3)
+    loop4 = FormattedComponent(LS336_LoopRO, "{self.prefix}", loop_number=4)
+
+    # same names as apstools.synApps._common.EpicsRecordDeviceCommonAll
+    scanning_rate = Component(EpicsSignal, "read.SCAN",kind=Kind.omitted)
+    process_record = Component(EpicsSignal, "read.PROC",kind=Kind.omitted)
+
+    read_all = Component(EpicsSignal, "readAll.PROC",kind=Kind.omitted)
+# TODO: serial = Component(AsynRecord, "serial")
+    serial = Component(MyAsynRecord, "serial",kind=Kind.omitted)
+
+lakeshore_336 = LS336Device("4idd:LS336:TC3:", name="lakeshore 360", labels=["Lakeshore"])
+
+####### LAKESHORE 340 #########
```

Comment:
good place to separate into a lakeshore340.py file

Reply:
move teh class definition into the new file and then  import & create instances in this file as needed.